src/main/perf/Indexer.java

```diff
@@ -597,4 +606,17 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
     }
     System.out.println("\nIndexer: " + (threads.getBytesIndexed()/1024./1024./1024./(indexingTime/3600000.)) + " GB/hour plain text");
   }
+
+  private static long countUniqueTerms(IndexWriter iw, String fld) throws IOException {
+    long total = 0;
+    try (IndexReader reader = DirectoryReader.open(iw)) {
+      for (LeafReaderContext ctx : reader.leaves()) {
+        long size = ctx.reader().terms(fld).size();
+        if (size > 0) {
```

Comment:
Ahh, this API is allowed to return -1 due to ancient codecs that did not implement this statistic, but it should never happen going forward.  But the safety is harmless, so I'll push!